### PR TITLE
Feature/coordinator admin actions

### LINKED ISSUE #268: Admin site: Actions with `Coordinator` that create log events with auto status change
**Please remove this `situational_status` because it duplicates the `project_status` with the same name**: 
```py
    PENDING = "pending", "Completed registration, waiting for onboarding"
```

Allow admins to create log events and set statuses for coordinators from the admin site.

The way I see it:
- admin chooses a log event type from a drop-down list
- a log event of this type gets created with current UTC time
- status (in all or most cases `project_status`) is set with the exact same timestamp as the date and time of creation of log event

Log event types and statuses are connected with rules:
- `APPLIED` -> `project_status=PENDING, situational_status=""`
- `JOINED` -> `project_status=PENDING, situational_status=""`
- `STARTED_ONBOARDING` -> `project_status=WORKING_BELOW_THRESHOLD, situational_status=ONBOARDING`
- `GONE_ON_LEAVE` -> `project_status=ON_LEAVE, situational_status=""`
- `LEFT_PREMATURELY` -> `project_status=LEFT_PREMATURELY, situational_status=""`
- `FINISHED_AND_LEAVING` -> `project_status=FINISHED_LEFT, situational_status=""`
- `FINISHED_AND_STAYING` -> `project_status=FINISHED_STAYS, situational_status=""`
- `EXPELLED` -> `project_status=BANNED, situational_status=""`
- `ACCESS_REVOKED` -> `project_status=REMOVED, situational_status=""`

The following events should trigger the same logic: choosing `project_status` between `WORKING_BELOW_THRESHOLD`, `WORKING_OK` or `WORKING_LIMIT_REACHED`, depending on number of groups, `situational_status=""`:
- `TRANSFER_COMPLETED`
- `TOOK_TRANSFERRED_GROUP`
- `RETURNED_FROM_LEAVE`
- `GROUP_ABORTED`
- `GROUP_FINISHED`
- `TOOK_NEW_GROUP` (should be triggered automatically from the dashboard, but let's include for the sake of completeness)
- `GROUP_STARTED_CLASSES` (should be triggered automatically from the dashboard, but let's include for the sake of completeness)

The following events should trigger no change in `project_status` or `situational_status`:
- `REQUESTED_TRANSFER`
- `TRANSFER_CANCELED`

### PR DESCRIPTION
**Pull request by Roxe322**
Closes #268

### CODE REVIEW

File: api/processors/auxil/log_event_creator.py [hunk 1]
```diff
@@ -52,3 +65,87 @@ def create(  # noqa: PLR0913
...
         )
+
+
+class CoordinatorAdminLogEventCreator:
+    @staticmethod
+    def _get_working_project_status(coordinator: Coordinator) -> CoordinatorProjectStatus:
+        group_count = coordinator.groups.count()
+        if group_count < CoordinatorGroupLimit.MIN:
+            return CoordinatorProjectStatus.WORKING_BELOW_THRESHOLD
+        if CoordinatorGroupLimit.MAX > group_count >= CoordinatorGroupLimit.MIN:
```
Comment (lemontree210) [inline api/processors/auxil/log_event_creator.py:76]:
```suggestion
        if CoordinatorGroupLimit.MIN <= group_count < CoordinatorGroupLimit.MAX:
```

File: api/processors/auxil/log_event_creator.py [hunk 2]
```diff
@@ -52,3 +65,87 @@ def create(  # noqa: PLR0913
...
+            return CoordinatorProjectStatus.WORKING_OK
+        if group_count >= CoordinatorGroupLimit.MAX:
+            return CoordinatorProjectStatus.WORKING_LIMIT_REACHED
+
+        raise ValueError(f"Unexpected group count: {group_count}, coordinator: {coordinator}")
```
Comment (lemontree210) [inline api/processors/auxil/log_event_creator.py:81]: Maybe `NotImplementedError` and include limits in the error message?

File: api/processors/auxil/log_event_creator.py [hunk 3]
```diff
@@ -52,3 +65,87 @@ def create(  # noqa: PLR0913
...
+                | CoordinatorLogEventType.GROUP_FINISHED
+                | CoordinatorLogEventType.TOOK_NEW_GROUP
+                | CoordinatorLogEventType.GROUP_STARTED_CLASSES
+            ):
+                return cls._get_working_project_status(coordinator), ""
+            case (
+                CoordinatorLogEventType.REQUESTED_TRANSFER
+                | CoordinatorLogEventType.TRANSFER_CANCELED
+            ):
+                # None means that status should not be changed
```
Comment (lemontree210) [inline api/processors/auxil/log_event_creator.py:123]: maybe this should be in the dosctring

**Comment by lemontree210** [comment]
Oh yes, I think that adding tests for log event creator would be a good idea. So many cases, easy to miss a mistake.

File: api/processors/auxil/log_event_creator.py (hunk 2 above)
Comment (Roxe322) [reply api/processors/auxil/log_event_creator.py]: Done

File: api/processors/auxil/log_event_creator.py (hunk 3 above)
Comment (Roxe322) [reply api/processors/auxil/log_event_creator.py]: Done

**Comment by Roxe322** [comment]
> Oh yes, I think that adding tests for log event creator would be a good idea. So many cases, easy to miss a mistake.

I thought that it will be mostly copy-paste of get_statuses_by_event_type, so decided to skip it. But can add them if it's needed :)

**Review by lemontree210 — APPROVED** [review]
LGTM

**Pull request merged**